dataprovider/baseadapter.go

```diff
@@ -41,8 +41,8 @@ type BaseAdapter struct {
 type BaseData struct {
 	Public    bool
 	ReadOnly  bool
-	Value     interface{}
 	subscribe bool
+	Value     interface{}
```

Comment:
Not so good. Mixed private and public types.

Reply:
commit was dropped due to  lint maligned becomes depricated